chasm/field_internal.go

```diff
@@ -28,14 +28,18 @@ func (fi fieldInternal) isEmpty() bool {
 }
 
 func (fi fieldInternal) value() any {
-	if fi.node == nil {
+	// Deferred pointers are special-cased, since their serialized nodes are
+	// initialized as regular persistable pointers.
+	if fi.node == nil || fi.ft == fieldTypeDeferredPointer {
```

Comment:
nit: you can add comment re. when deferredPointers will have non-nil `fi.node`? Took me some time to realize that's only temporarily after syncSubField and before resolving the deferred pointers